### PR TITLE
Add optional parameter to use as a default

### LINKED ISSUE #138: Allow to provide an optional default value
For some cases it might be useful to be able to provide a default value if no initial state is provided. This would help to simplify the code a bit as you could avoid requiring to catch errors:

```
let myState
try {
  myState = loadState('app', 'key')
} catch (e) {
  myState = 1234
}
```

would become:

```
let myState = loadState('app', 'key', 1234)
```

Any concerns @nextcloud/javascript ? Otherwise I'd prepare a PR.

### PR DESCRIPTION
**Pull request by juliushaertl**
Fixes #138

### CODE REVIEW

**Comment by juliushaertl** [comment]
@ChristophWurst Good to merge? fixupbot is confused

File: lib/index.ts [hunk 1]
```diff
@@ -1,17 +1,21 @@
...
+            return fallback
+        }
         throw new Error(`Could not find initial state ${key} of ${app}`)
     }
 
     try {
         return JSON.parse(atob(elem.value))
     } catch (e) {
-       throw new Error(`Could not parse initial state ${key} of ${app}`)
+        throw new Error(`Could not parse initial state ${key} of ${app}`)
```
Comment (ChristophWurst) [inline lib/index.ts:19]: well well well

**Review by ChristophWurst — COMMENTED** [review]
I'm fine with the change, but not this whitespace :P

**Comment by ChristophWurst** [comment]
Just merge once that is resolved :)

File: lib/index.ts (hunk 1 above)
Comment (skjnldsv) [reply lib/index.ts:19]: 🐋 🐋 🐋

Comment (juliushaertl) [reply lib/index.ts]: :whale: it is a valid fix whitespace fix, but removed it to keep changes clean.

**Pull request merged**